dbms/src/Storages/MergeTree/MergeTreeDataFormatVersion.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <common/strong_typedef.h>
+
+namespace DB
+{
+
+STRONG_TYPEDEF(UInt32, MergeTreeDataFormatVersion);
+
+const MergeTreeDataFormatVersion MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING {1};
```

Comment:
It can fail at link time if we ever take an address of this constant (example: debug build and passing it to std::max).

Reply:
Global constants should be OK (they have internal linkage by default). The problem may be with static const class members, but it is not the case here.